std/string.d

```diff
@@ -5993,6 +6014,17 @@ bool isNumeric(S)(S s, bool bAllowSep = false) if (isSomeString!S)
     }
 }
 
+// test random access ranges
+@safe @nogc pure nothrow unittest
+{
+    import std.uni : byCodePoint;
+    assert("123".byCodePoint.isNumeric());
+    assert("123UL".byCodePoint.isNumeric());
```

Comment:
`"foo".byCodePoint` is not a random-access range. It's just the same `string` again.